terraform/resource_provider_mock.go

```diff
@@ -56,6 +56,8 @@ type MockResourceProvider struct {
 	ReadDataDiffFn                 func(*InstanceInfo, *ResourceConfig) (*InstanceDiff, error)
 	ReadDataDiffReturn             *InstanceDiff
 	ReadDataDiffReturnError        error
+	TestResetCalled                bool
+	TestResetError                 error
```

Comment:
Makes sense. Removed the fields from terraform, and wrapped the mock provider with a TestProvider implementation.